test/nodes/test_prompt_node.py

```diff
@@ -245,9 +245,40 @@ def test_stop_words(prompt_model):
     if prompt_model.api_key is not None and not is_openai_api_key_set(prompt_model.api_key):
         pytest.skip("No API key found for OpenAI, skipping test")
 
+    # test stop words for both HF and OpenAI
+    # set stop words in PromptNode
     node = PromptNode(prompt_model, stop_words=["capital", "Germany"])
+
+    # with default prompt template and stop words set in PN
     r = node.prompt("question-generation", documents=["Berlin is the capital of Germany."])
-    assert r[0] == "What is the"
+    assert r[0] == "What is the" or r[0] == "What city is the"
+
+    # with default prompt template and stop words set in kwargs (overrides PN stop words)
+    r = node.prompt("question-generation", documents=["Berlin is the capital of Germany."], stop_words=None)
+    assert "capital" in r[0] or "Germany" in r[0]
+
+    # simple prompting
+    r = node("Given the context please generate a question. Context: Berlin is the capital of Germany.; Question:")
+    assert len(r[0]) > 0 and "capital" not in r[0] and "Germany" not in r[0]
```

Comment:
```suggestion
    assert len(r[0]) > 0
    assert "capital" not in r[0]
    assert "Germany" not in r[0]
```

I would unroll this into multiple so in case it fails it's clear at a glance why.

Reply:
Ok, I'll do that for all checks in that test! What about other stuff @silvanocerza ?

Reply:
Ok we can unroll only in this instance.

Reply:
Am not sure doing the same for other `assert`s is a good idea, they're a bit different from this one.

Take the next one for example `assert "capital" in r[0] or "Germany" in r[0]`, that `or` means to me that we're fine whether there is `capital` or `Germany` or both. In the case there's neither it would fail and that's good.

If we unroll it like this:

```
assert "capital" in r[0]
assert "Germany" in r[0]
```

The first might pass and the second fail and break the test, but that's not what we want right?

The rest is good by the way. 👍

Reply:
Yes, good reasoning breakup - thank you @silvanocerza